File: app_be/booking-service/src/main/java/com/medibook/booking/dto/TimeSlotDto.java
package com.medibook.booking.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.medibook.common.enums.SlotStatus;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

/**
 * DTO cho TimeSlot response
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TimeSlotDto {
    private UUID id;
    private UUID doctorId;
    
    // Thông tin bác sĩ (để hiển thị trên Admin)
    private String doctorName;
    private String doctorAvatar;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate date;

    @JsonFormat(pattern = "HH:mm:ss")
    private LocalTime startTime;

    @JsonFormat(pattern = "HH:mm:ss")
    private LocalTime endTime;

    private Boolean isAvailable;

    @Builder.Default
    private SlotStatus status = SlotStatus.PENDING;
}

File: app_admin/lib/features/schedules/data/dto/time_slot_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'time_slot_dto.freezed.dart';
part 'time_slot_dto.g.dart';

@freezed
class TimeSlotDto with _$TimeSlotDto {
  const factory TimeSlotDto({
    required String id,
    required String doctorId,
    String? doctorName,
    String? doctorAvatar,
    required String date,
    required String startTime,
    required String endTime,
    @Default(true) bool isAvailable,
    @Default('PENDING') String status,
  }) = _TimeSlotDto;

  factory TimeSlotDto.fromJson(Map<String, dynamic> json) =>
      _$TimeSlotDtoFromJson(json);
}
